packages/open-payments/README.md

```diff
@@ -0,0 +1,197 @@
+| `keyId`             | The key identifier of the given private key and the corresponding public JWK document.                                                                                                                                                                                                                                                                                                      |
+
+> **Note**
+>
+> To simplify EdDSA-Ed25519 key provisioning and JWK generation, you can use the `parseOrProvisionKey` and `generateJwk` methods from the [`http-signature-utils`](../http-signature-utils/README.md#usage) package.
+
+## Example
+
+As mentioned previously, Open Payments APIs can facilitate a payment between two parties.
+
+For example, say Alice wants to purchase a $50 product from a merchant called Shoe Shop on the Amazon Marketplace. If both parties have Open Payments enabled wallets, where Alice's payment pointer is `https://cloud-nine-wallet/alice`, and Shoe Shop's is `https://happy-life-bank/shoe-shop`, requests during checkout from Amazon's backend would look like this using the client:
+
+1. Create an Open Payments client
+
+In this case, since Amazon Marketplace wants to make requests that require authorization, it will need to create an `AuthenticatedClient`:
+
+```ts
+import { createAuthenticatedClient } from '@interledger/open-payments'
+
+const client = await createAuthenticatedClient({
+  paymentPointerUrl: 'https://amazon.com/usa',
+  keyId: KEY_ID,
+  privateKey: PRIVATE_KEY
+  // The public JWK with this key (and keyId) would be available at https://amazon.com/usa/jwks.json
+})
+```
+
+2.  Get `PaymentPointers`
+
+Grab the payment pointers of the parties:
+
+```ts
+const shoeShopPaymentPointer = await client.paymentPointer.get({
+  url: 'https://happy-life-bank/shoe-shop'
+})
+
+const customerPaymentPointer = await client.paymentPointer.get({
+  url: 'https://cloud-nine-wallet/alice'
+})
+```
+
+3. Create `IncomingPayment`
+
+Amazon's backend gets a grant to create an `IncomingPayment` on the merchant's wallet:
+
+```ts
+const incomingPaymentGrant = await client.grant.request(
+  { url: shoeShopPaymentPointer.authServer },
+  {
+    access_token: {
+      access: [
+        {
+          type: 'incoming-payment',
+          actions: ['read-all', 'create']
+        }
+      ]
+    }
+  }
+)
+
+const incomingPayment = await client.incomingPayment.create(
+  {
+    paymentPointer: shoeShopPaymentPointer.authServer,
+    accessToken: incomingPaymentGrant.access_token.value
+  },
+  {
+    incomingAmount: {
+      assetCode: 'USD',
+      assetScale: 2,
+      value: '5000'
+    },
+    description: 'Purchase at Shoe Shop',
+    externalRef: '#INV2022-8363828'
+  }
+)
+```
+
+4. Create `Quote`
+
+Then, it'll get a grant to create a `Quote` on Alice's payment pointer, which will give the amount it'll cost Alice to make the payment (with the ILP fees + her wallet's fees)
```

Comment:
If we don't change it, quote grant requests are currently also interactive. The way quoting works right now, I think we should be able to make it non-interactive. But that may not be the case in the future with multiple payment methods.

Reply:
I can make an issue to have quote grant interactivity behind a flag (similar to `INCOMING_PAYMENT_INTERACTION`) and disable it by default

Reply:
- #1215

Reply:
Do you want to get #1215 in first before merging this or do you want to mention that quotes are interactive right now and get this in first?

Reply:
I've looked, it doesn't seem to be too big of a lift so I'll try to get that in first